src/signalrclient/connection_impl.cpp

```diff
@@ -577,21 +599,31 @@ namespace signalr
     void connection_impl::stop(std::function<void(std::exception_ptr)> callback, std::exception_ptr exception) noexcept
     {
         m_stop_error = exception;
-        m_logger.log(trace_level::info, "stopping connection");
+        m_logger.log(trace_level::info, "closing connection");
 
         shutdown(callback);
     }
 
     // This function is called from the dtor so you must not use `shared_from_this` here (it will throw).
-    void connection_impl::shutdown(std::function<void(std::exception_ptr)> callback)
+    void connection_impl::shutdown(std::function<void(std::exception_ptr)> callback, bool is_dtor)
     {
         {
             std::lock_guard<std::mutex> lock(m_stop_lock);
-            m_logger.log(trace_level::info, "acquired lock in shutdown()");
+            m_logger.log(trace_level::verbose, "acquired lock in shutdown()");
+            auto logger = m_logger;
+            auto _ = callback_on_destruct([logger]()
+                {
+                    logger.log(trace_level::verbose, "released lock in shutdown()");
+                });
 
             const auto current_state = get_connection_state();
             if (current_state == connection_state::disconnected)
             {
+                // don't log if already disconnected and shutdown called from dtor, it's just noise
+                if (!is_dtor)
```

Comment:
Or change the log to say it's from the dtor and make it verbose in that case?

Reply:
Considering that "acquired lock in shutdown()" and "released lock in shutdown()" will still be logged, I think it's better to log something at verbose in this case.

Reply:
Yeah, I've been mulling it over and think I'll have an additional log of "calling shutdown() from dtor" and then change the "already stopped" log to verbose in the case of it coming from the dtor.